src/generic/Matrix.jl

```diff
@@ -2371,6 +2323,104 @@ function can_solve_left_reduced_triu(r::AbstractAlgebra.MatElem{T},
    return true, x
 end
 
+@doc Markdown.doc"""
+    can_solve_with_solution(a::AbstractAlgebra.MatElem{S}, b::AbstractAlgebra.MatElem{S}; side = :right) where S <: RingElement
+> Given two matrices $a$ and $b$ over the same ring, tries to solve $ax = b$
+> if `side` is `:right` or $xa = b$ if `side` is `:left`. In either case,
+> returns a tuple `(flag, x)`. If a solution exists, `flag` is set to true and
+> `x` is a solution. If no solution exists, `flag` is set to false and `x`
+> is arbitrary. If the dimensions of $a$ and $b$ are incompatible, an exception
+> is raised.
+"""
+function can_solve_with_solution(a::AbstractAlgebra.MatElem{S}, b::AbstractAlgebra.MatElem{S};
+                                                                                side = :right) where S <: RingElement
```

Comment:
```suggestion
                                                                                side::Symbol = :right) where S <: RingElement
```
Also, as you have a newline here, you could align `side` more on the left, with `a` for example.

Reply:
I wanted to emulate the formatting in [Hecke.jl](https://github.com/thofma/Hecke.jl/blob/master/src/Misc/Matrix.jl#L1822-L1823), but now I see that in AbstractAlgebra.jl, the function signature is generally all in the same line. I personally prefer it this way too, so if it's ok I would change it like this:

```
- function can_solve_with_solution(a::AbstractAlgebra.MatElem{S}, b::AbstractAlgebra.MatElem{S};
-                                                                                 side = :right) where S <: RingElement
+ function can_solve_with_solution(a::AbstractAlgebra.MatElem{S}, b::AbstractAlgebra.MatElem{S}; side::Symbol = :right) where S <: RingElement 
```

(Sorry about the formatting, I can't figure out how to do it with "suggestion")

Reply:
Sure do it as you prefer. I don't think there is convention in AbstractAlgebra regarding line lenghts, I like it not too long though (.e.g less than 92 characters) as it makes it more difficult to read when one's screen is not that wide.